### PR TITLE
checkpoint-*: add pending writes to .list()

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/__init__.py [hunk 1]
```diff
@@ -318,7 +318,9 @@ def list(
         ORDER BY checkpoint_id DESC"""
         if limit:
             query += f" LIMIT {limit}"
-        with self.cursor(transaction=False) as cur:
+        with self.cursor(transaction=False) as cur, self.cursor(
```
Comment (nfcampos) [inline libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/__init__.py:321]: two cursors?

File: libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/aio.py [hunk 2]
```diff
@@ -346,6 +346,14 @@ async def alist(
                 checkpoint,
                 metadata,
             ) in cursor:
+                writes_cur = await self.conn.execute(
```
Comment (nfcampos) [inline libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/aio.py:349]: this cursor is never closed it seems

Comment (vbarda) [reply libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/aio.py:349]: let me adjust!

File: libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/__init__.py (hunk 1 above)
Comment (vbarda) [reply libs/checkpoint-sqlite/langgraph/checkpoint/sqlite/__init__.py:321]: not sure what's a better way to do it other than just put everything into a single query and join aggregated writes (not sure if doable in sqlite)

**Comment by nfcampos** [comment]
Superseded by #1436

**Pull request closed without merging**